File: js/link.js
/**
 * Link generator - creates shareable URLs
 * Single responsibility: URL construction
 */

const BASE_URL = 'https://zephr.io';

/**
 * Generate shareable link
 * @param {string} secretId - Secret identifier from API
 * @param {string} keyString - Versioned key string
 * @param {boolean} splitUrlMode - Whether to split URL and key
 * @returns {{mode: string, url?: string, key?: string, fullLink?: string}}
 */
export function generateLink(secretId, keyString, splitUrlMode) {
    if (splitUrlMode) {
        return {
            mode: 'split',
            url: `${BASE_URL}/secret/${secretId}`,
            key: keyString
        };
    }

    return {
        mode: 'standard',
        fullLink: `${BASE_URL}/secret/${secretId}#${keyString}`
    };
}

File: python/src/zephr/link.py
"""
Link generator — creates shareable URLs.
Single responsibility: URL construction only.
"""

_BASE_URL = "https://zephr.io"


def generate_link(
    secret_id: str,
    key_string: str,
    split: bool = False,
) -> dict:
    """Generate a shareable link for a secret.

    Args:
        secret_id: Secret identifier from the API.
        key_string: Versioned key string (e.g., "v1.<base64url>").
        split: Whether to return URL and key separately.

    Returns:
        Dict with 'mode' and either 'full_link' or 'url'+'key'.
    """
    if split:
        return {
            "mode": "split",
            "url": f"{_BASE_URL}/secret/{secret_id}",
            "key": key_string,
        }

    return {
        "mode": "standard",
        "full_link": f"{_BASE_URL}/secret/{secret_id}#{key_string}",
    }
